lib/segment/src/index/payload_index_base.rs

```diff
@@ -12,10 +12,19 @@ use crate::index::field_index::{CardinalityEstimation, PayloadBlockCondition};
 use crate::json_path::JsonPath;
 use crate::payload_storage::FilterContext;
 use crate::types::{
-    Filter, Payload, PayloadFieldSchema, PayloadKeyType, PayloadKeyTypeRef, PayloadSchemaType,
-    SeqNumberType,
+    Filter, Payload, PayloadFieldSchema, PayloadKeyType, PayloadKeyTypeRef, SeqNumberType,
 };
 
+pub enum BuildIndexResult {
+    /// Index was built
+    Built(Vec<FieldIndex>),
+    /// Index was already built
+    AlreadyBuilt,
+    /// Field Index already exists, but incompatible schema
+    /// Requires extra actions to remove the old index.
+    IncompatibleSchema,
+}
```

Comment:
This looks too similar to `BuildFieldIndexResult`, is it possible to keep just one? or what is the reason for a separate enum?

Reply:
Similar, but the other one have one extra field.